server/manager/src/main/java/org/apache/accumulo/manager/metrics/fate/FateMetrics.java

```diff
@@ -141,46 +197,51 @@ protected void prepareMetrics() {
   private void recordValues() {
 
     // update individual gauges that are reported.
-    currentFateOps.set(metricValues.getCurrentFateOps());
-    zkChildFateOpsTotal.set(metricValues.getZkFateChildOpsTotal());
-    zkConnectionErrorsTotal.set(metricValues.getZkConnectionErrors());
+    // currentFateOps.set(metricValues.getCurrentFateOps());
+    // zkChildFateOpsTotal.set(metricValues.getZkFateChildOpsTotal());
+    // zkConnectionErrorsTotal.set(metricValues.getZkConnectionErrors());
 
     // the number FATE Tx states (NEW< IN_PROGRESS...) are fixed - the underlying
     // getTxStateCounters call will return a current valid count for each possible state.
-    Map<String,Long> states = metricValues.getTxStateCounters();
-
-    states.forEach((key,
-        value) -> fateTypeCounts.computeIfAbsent(key,
-            v -> super.getRegistry().newGauge(metricNameHelper(FATE_TX_STATE_METRIC_PREFIX, key),
-                "By transaction state count for " + key, value))
-            .set(value));
-
+    // Map<String,Long> states = metricValues.getTxStateCounters();
+
+    /*
+     * REPLACED WITH MICROMETER GAUGE IN CONSTRUCTOR states.forEach((key, value) ->
+     * fateTypeCounts.computeIfAbsent(key, v ->
+     * super.getRegistry().newGauge(metricNameHelper(FATE_TX_STATE_METRIC_PREFIX, key),
+     * "By transaction state count for " + key, value)) .set(value));
+     */
     // the op types are dynamic and the metric gauges generated when first seen. After
     // that the values need to be cleared and set to any new values present. This is so
     // that the metrics system will report "known" values once seen. In operation, the
     // number of types will be a fairly small set and should populate a consistent set
     // with normal operations.
 
+    // DONT NEED THIS SINCE MICROMETER GAUGE DOESNT NEED TO BE SET
```

Comment:
I think I understand, I'm not sure this same approach will work with micrometer gauges since the value will just be recomputed each time the gauge is read so even if we set it to 0, 0 will not be returned once we request the gauges value.

I'll have to look into a solution for this.

Reply:
Using the example from their docs (https://micrometer.io/docs/concepts#_gauges) one approach my be to create AtomicIntegers when first seen, store them in the map, register those with registry.gauge(key, atomic_value_ref) - then the metrics updates would 1) set each atomic_value_ref in the map to 0, then process the current counts and set the current values.  Creating the AtomicIntegers and storing them in a map (keyed by opname) would also satisfy the need to keep a strong reference to the values.

This assumes that something like the following would be valid

```
AtomicInteger x = new AtomicInteger(0);
...
registry.gauge(key, x);
...
x.set(value)

(x would be stored in the map, keyed with optype name)
```

works and you don't need the reference returned from registry.gauge(....)

Reply:
@EdColeman, I did my best to address this in 85f71d3.